### PR TITLE
extending the initrd_sdk

### PR DESCRIPTION
**Pull request by Rozzii**
This commit adds:

- New script used to avoid race conditions while running root switching
- New systemd unit file that is used to run the previously introduced unlock-mount-luks.sh
- Documentation about the new files

### CODE REVIEW

File: jenkins/image_building/initrd_sdk/verify-realroot.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+while true; do
+    if [ -e "/realroot/bin" ]; then
+        printf "INFO: Realroot mount point is present.\n"
+        break
+    else
+        printf "INFO: Waiting for realroot!\n"
+        # Introduce a 1-second delay using the read command
+        # sleep might not be available but this way we stress
+        # the CPU less
+        read -r -t 1 < "/dev/null"
```
Comment (mboukhalfa) [inline jenkins/image_building/initrd_sdk/verify-realroot.sh:17]: I tried this seems not waiting on my terminal ! is tested somehow ?

**Review by mboukhalfa — COMMENTED** [review]
If I understand correctly the verify-realroot.sh is introduced but not used so far ?

File: jenkins/image_building/initrd_sdk/verify-realroot.sh (hunk 1 above)
Comment (Rozzii) [reply jenkins/image_building/initrd_sdk/verify-realroot.sh:17]: yeah I have tested it it was waiting for me although I was testing this in a initrd.

Comment (Rozzii) [reply jenkins/image_building/initrd_sdk/verify-realroot.sh]: now it is good, I have done a modification after I have tested, nice catch.

**Pull request merged**